### PR TITLE
[Core] adding Testing::GetDefaultDataCommunicator

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
After #8612, making C++ consistent with Python

Also fixed the deprecation warnings in the Core
I plan to fix the deprecation warnings in the remaining Core-C++-Tests after this PR

### CODE REVIEW

File: kratos/testing/testing.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
```
Comment (rubenzorrilla) [inline kratos/testing/testing.cpp]:
```suggestion
//  License:         BSD License
//                   Kratos default license: kratos/license.txt
//
//  Main authors:    Pooyan Dadvand
```

File: kratos/testing/distributed_test_case.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ std::string DistributedTestCase::Info() const
 void DistributedTestCase::CheckRemoteFailure()
 {
     bool success_on_this_rank = GetResult().IsSucceed();
-    const DataCommunicator& r_comm = DataCommunicator::GetDefault();
+    const DataCommunicator& r_comm = ParallelEnvironment::GetDefaultDataCommunicator();
```
Comment (rubenzorrilla) [inline kratos/testing/distributed_test_case.cpp:62]: Why not using the one from `testing` namespace in here?

Comment (rubenzorrilla) [reply kratos/testing/distributed_test_case.cpp:62]: (would be the same but seems more consistent)

Comment (philbucher) [reply kratos/testing/distributed_test_case.cpp:62]: I would like to use the `Testing::GetDefaultDataCommunicator()` exclusively in the tests
This is not a test, hence I would leave it as is, as it is IMO one of the few valid usecases of `ParallelEnvironment::GetDefaultDataCommunicator`

If you insist I can change it but I would prefer to leave it as is

Comment (rubenzorrilla) [reply kratos/testing/distributed_test_case.cpp:62]: No no, it was just a suggestion. Approving.

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**